algorithms/a3c/main.py

```diff
@@ -40,80 +54,276 @@
+    args.num_backprops = 0
+    if not args.experiment_id:
+        args.experiment_id = datetime.datetime.now().strftime("%Y-%m-%d-") \
+                                                                     + str(uuid.uuid4())
+    args.experiment_path = os.path.join(os.path.abspath(os.path.join(os.path.dirname(__file__),
+                           '..', '..', 'experiments', args.experiment_id)))
+    args.checkpoint_path = os.path.join(args.experiment_path, 'checkpoints')
+    args.tensorboard_path = os.path.join(args.experiment_path, 'tensorboard_logs')
 
-    torch.manual_seed(args.seed)
+    # Create environment from choices of Atari, ViZDoom and our ai2thor wrapper with tasks
     if args.atari:
         env = create_atari_env(args.atari_env_name)
-        args.frame_dim = 42  # fixed to be 42x42 in envs.py _process_frame42()
+        args.resolution = (42, 42)  # fixed to be 42x42 for _process_frame42() in envs.py
+    elif args.vizdoom:
+        # many more dependencies required for VizDoom so therefore import here
+        from algorithms.a3c.env_vizdoom import GroundingEnv
+
+        if args.evaluate == 0:
+            args.use_train_instructions = 1
+        elif args.evaluate == 1:
+            args.use_train_instructions = 1
+            args.num_processes = 0
+        elif args.evaluate == 2:
+            args.use_train_instructions = 0
+            args.num_processes = 0
+        else:
+            assert False, "Invalid evaluation type"
+
+        env = GroundingEnv(args)
+        env.game_init()
+        args.resolution = (args.frame_width, args.frame_height)
     else:
-        args.config_dict = {'max_episode_length': args.max_episode_length}
-        env = AI2ThorEnv(config_dict=args.config_dict)
-        args.frame_dim = env.config['resolution'][-1]
-    shared_model = ActorCritic(env.observation_space.shape[0], env.action_space.n, args.frame_dim)
-    shared_model.share_memory()
+        # if you resume a checkpoint and if args.resume_latest_config == True, changes to the file
+        # will be overwritten by the latest_config.json file. Set to False if you want to change
+        # config settings in the config file after resuming and in the middle of training
+        args.last_config_resume_path = os.path.join(args.experiment_path, 'latest_config.json')
+        args.config_dict = {}
+        if args.resume_latest_config:
+            print('args.resume_latest_config is set on')
+            # read last_config.json if it exists, else create it below
+            if os.path.exists(args.last_config_resume_path):
+                print('Folder for this args.experiment_id "{}" and latest_config file existed'
+                      .format(args.experiment_id))
+                with open(args.last_config_resume_path, 'r') as f:
+                    print('Loading config_dict from file: {}'.format(args.last_config_resume_path))
+                    print('Therefore changes to args.config_file_name won\'t affect the env.\n'
```

Comment:
Rephrase it for clarity
```suggestion
                    print('Changes to args.config_file_name won\'t affect the env.\n' 
                    'Set to args.resume_latest_config to 0 to override args.config_file with xxx params'
```